curriculum/challenges/english/25-front-end-development/workshop-spam-filter/64220fb017c57d20612de8b8.md

```diff
@@ -0,0 +1,195 @@
+---
+id: 64220fb017c57d20612de8b8
+title: Step 16
+challengeType: 0
+dashedName: step-16
+---
+
+# --description--
+
+Now that you have your capture group, you can mark the entire pattern as an optional match. The `?` quantifier matches zero or one occurrence of the preceding character or group. For example, the regular expression `/colou?r/` matches both `color` and `colour`, because the `u` is optional.
+
+Mark your capture group as optional.
+
+# --hints--
+
+Your `dollarRegex` should use the `?` quantifier.
+
+```js
+assert.match(dollarRegex.source, /\?/);
+```
+
+Your `(hundred|thousand|million|billion)` capture group should be optional.
+
+```js
+assert.match(dollarRegex.source, /\(hundred\|thousand\|million\|billion\)\?/);
+```
+
+Your `dollarRegex` should match `1000 dollars`.
+
+```js
+const dollarStr = '1000 dollars'
+assert.deepEqual(dollarStr.match(dollarRegex), dollarStr.match(/[0-9]+ (hundred|thousand|million|billion)? dollars/i));
+```
```

Comment:
this doesn't match, you need two spaces in the string to be matched by this regex.

Reply:
I'm just going to remove this test. In the following step, it's mentioned that this won't match `10 dollars`. So it shouldn't be an issue.

Reply:
Seems reasonable!